addons/sale_loyalty/models/sale_order.py

```diff
@@ -43,6 +44,70 @@ def _compute_reward_total(self):
                     reward_amount -= line.product_id.lst_price * line.product_uom_qty
             order.reward_amount = reward_amount
 
+    def _add_loyalty_history_line(self):
+        for order in self:
+            points_per_coupon = defaultdict(dict)
+            for coupon_point in order.coupon_point_ids:
+                points_per_coupon[coupon_point.coupon_id]['issued'] = coupon_point.points
+            for line in order.order_line:
+                if not line.coupon_id:
+                    continue
+                points_per_coupon[line.coupon_id]['cost'] = line.points_cost
+
+            for coupon, point_dict in points_per_coupon.items():
+                cost = point_dict.get('cost', 0.0)
+                issued = point_dict.get('issued', 0.0)
+                order.env['loyalty.history'].create({
+                    'card_id': coupon.id,
+                    'order_id': '%s,%i' % (order._name, order.id),
+                    'description': order._get_history_line_description(),
+                    'used': cost,
+                    'issued': issued,
+                    "new_balance": coupon.points + issued - cost,
+                })
+
+    @api.depends('order_line', 'state')
+    def _compute_loyalty_data(self):
+        confirmed_so = self.filtered(lambda order: order.state == 'sale' and order.id)
+        (self - confirmed_so).loyalty_data = {}
+        if not confirmed_so:
+            return
+
+        order_list = ['sale.order,%i' % (order.id) for order in confirmed_so]
+        loyalty_data = self.env['loyalty.history'].sudo()._read_group(
+            domain=[
+                ('order_id', 'in', order_list)
+            ],
+            groupby=['order_id'],
+            aggregates=['new_balance:sum', 'issued:sum', 'used:sum'],
+        )
+        order_loyalty_mapping = {
+            data[0]: {
+                'total_old_balance': data[1] - data[2] + data[3],
+                'total_issued': data[2],
+                'total_cost': data[3],
+                'total_new_balance': data[1],
+            }
+            for data in loyalty_data
+        }
+        for order in confirmed_so:
+            order_reference = 'sale.order,%i' % (order.id)
+            if order_reference not in order_loyalty_mapping:
+                order.loyalty_data = {}
+                continue
+            coupons = order.coupon_point_ids.coupon_id
+            coupon_point_name = (coupons and coupons[0].point_name) or 'Points'
```

Comment:
```suggestion
            coupon_point_name = (len(coupons) == 1 and coupons.point_name) or _("Points")
```
Display coupon point name if there is only one coupon else points also translation